utils/fasm/src/fasm.h

```diff
@@ -0,0 +1,61 @@
+#ifndef FASM_H
+#define FASM_H
+
+#include <iostream>
+#include <list>
+#include <map>
+#include <ostream>
+#include <set>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "netlist_walker.h"
+
+class FasmWriterVisitor : public NetlistVisitor {
+    enum LutMode {
+      NO_LUT = 0,
+      LUT = 1,
+      SPLIT_LUT = 2,
+    };
+
+  public:
+      FasmWriterVisitor(std::ostream& f);
+
+  private:
+      void visit_top_impl(const char* top_level_name) override;
+      void visit_open_impl(const t_pb* atom) override;
+      void visit_atom_impl(const t_pb* atom) override;
+      void visit_clb_impl(ClusterBlockId blk_id, const t_pb* clb) override;
```

Comment:
Maybe some type of note might be useful as `visit_clb_impl` stands for `visit_cluster_block` rather than the "complex logic block".

Reply:
`s/cluster_block/tile/`